lib/pages/home/profile_card.dart

```diff
@@ -28,13 +29,28 @@ class ProfileCard extends StatelessWidget {
               Row(
                 crossAxisAlignment: CrossAxisAlignment.center,
                 children: <Widget>[
-                  const Padding(
-                    padding: EdgeInsets.symmetric(horizontal: 8),
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 8),
                     child: CircleAvatar(
                       radius: 40,
-                      child: Image(
-                          image: AssetImage(
-                              'assets/illustrations/undraw_profile_pic.png')),
+                      child: user != null
+                          ? FutureBuilder(
+                              future: authProvider.getProfilePicture(context),
+                              builder: (context, snapshot) {
+                                if (snapshot.hasData) {
+                                  return CircleImage(
```

Comment:
What's the point of CircleImage here? I think you can remove this and Image below, and instead of `child` use `backgroundImage` in `CircleAvatar`, it takes an `ImageProvider`.

Reply:
`ImageProvider<dynamic>` is not a subtype of `FutureBuilder<String>` and it will throw an exception.